### ISSUE TITLE
result of bert positional embed changed in 2.8

### ISSUE THREAD

**Issue by jianyizh** [intel/torch-xpu-ops#1811]
### 🐛 Describe the bug
when install xpu build pytorch but run on cpu, I have the following error. It will not happen on cuda

```python
import math
import torch

div_term = (torch.arange(0, 768, 2).float() * -(math.log(10000.0) / 768)).exp()
torch.save(div_term,"d0.pt")
```
Compare the result of 2.7 release and 2.8rc, they are different, can cause torchbench bert accuracy failed.

```
>>> import torch
>>> a=torch.load("d2.7.pt")
>>> b=torch.load("d2.8.pt")
>>> a==b
tensor([ True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True, **False**,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True, **False**,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True, **False**,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True, False,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True, False,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True, False,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True, False,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True, False,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True, False,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True])
```

### Versions

it should be a compiler issue when upgrade from 2025.0 to 2025.1

**Comment by CuiYifeng** [comment]
Need to check oneMKL with C++ case.

**Comment by CuiYifeng** [comment]
oneMKL mentioned that such accuracy change of `vmsExp` is expected.

**Comment by jianyizh** [comment]
I still cannot reproduce the model fail under my local env.

**Comment by pbielak** [comment]
Issue still persists (comparing output of PyTorch 2.7.0 vs current main/PyTorch 2.12):
```
>>> import torch

>>> a = torch.load("2_7.pt")
>>> b = torch.load("2_12.pt")

>>> a == b
tensor([ True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True, False,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True, False,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True, False,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True, False,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True, False,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True, False,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True, False,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True, False,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True, False,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True,  True,  True,  True,  True,  True,  True,
         True,  True,  True,  True])

>>> (a != b).nonzero()
tensor([[ 31],
        [ 47],
        [ 67],
        [109],
        [224],
        [262],
        [276],
        [309],
        [357]])

>>> a[31].item(), b[31].item()
(0.4754264||6527290344, 
 0.4754264||3547058105)
>>> a[47].item(), b[47].item()
(0.3239042||7589416504, 
 0.3239042||4609184265)
>>> a[67].item(), b[67].item()
(0.2004856||7652702332, 
 0.2004856||6162586212)
>>> a[109].item(), b[109].item()
(0.0732121||8401193619, 
 0.0732121||7656135559)
>>> a[224].item(), b[224].item()
(0.0046415||90174287558, 
 0.0046415||8970862627)
>>> a[262].item(), b[262].item()
(0.001865664||147771895, 
 0.001865664||031356573)
>>> a[276].item(), b[276].item()
(0.001333521||4462131262, 
 0.001333521||3297978044)
>>> a[309].item(), b[309].item()
(0.000604296||5105734766, 
 0.000604296||4523658156)
>>> a[357].item(), b[357].item()
(0.0001910952||851176262, 
 0.0001910952||7056571096)
```

**Comment by jianyizh** [comment]
We may need update cpu mkl-static to avoid cpu mkl link to dle. It's not related to gpu
